src/lib.rs

```diff
@@ -383,25 +382,17 @@ macro_rules! try_callback {
 /// provided buffer, up to a max of `len` bytes. Output is UTF-8 encoded
 /// and NUL terminated. Returns the number of bytes written before the NUL.
 #[no_mangle]
-pub extern "C" fn rustls_version(buf: *mut c_char, len: size_t) -> size_t {
-    ffi_panic_boundary! {
-        let write_buf: &mut [u8] = unsafe {
-            if buf.is_null() {
-                return 0;
-            }
-            slice::from_raw_parts_mut(buf as *mut u8, len as usize)
-        };
-        let version: String = format!(
-            "crustls/{}/rustls/{}",
-            env!("CARGO_PKG_VERSION"),
-            RUSTLS_CRATE_VERSION,
-        );
-        let version: &[u8] = version.as_bytes();
-        let len: usize = min(write_buf.len() - 1, version.len());
-        write_buf[..len].copy_from_slice(&version[..len]);
-        write_buf[len] = 0;
-        len
-    }
+pub extern "C" fn rustls_version() -> rustls_str<'static> {
+    return rustls_str::from_str_unchecked(crate::constants::RUSTLS_FFI_VERSION);
+}
+
+#[test]
+fn test_rustls_version() {
+    // very rough check that the version number is being interpolated into the 
+    // variable
+    assert!(crate::constants::RUSTLS_FFI_VERSION.contains("/0."));
```

Comment:
We could also e.g. write CARGO_PKG_VERSION to constants.rs and then check that FFI_VERSION contains CARGO_PKG_VERSION.